logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -190,14 +192,17 @@ abstract class UserSessionScopeCommon(
             userSearchApiWrapper
         )
 
+    val persistMessage: PersistMessageUseCase
+        get() = PersistMessageUseCaseImpl(messageRepository, conversationRepository, userId)
+
     private val callRepository: CallRepository by lazy {
         CallDataSource(
             callApi = authenticatedDataSourceSet.authenticatedNetworkContainer.callApi,
             conversationRepository = conversationRepository,
             userRepository = userRepository,
             teamRepository = teamRepository,
             timeParser = timeParser,
-            messageRepository = messageRepository
+            persistMessage = persistMessage
```

Comment:
Hmm but now we are injecting use cases into repositories, or is the only place and the CallDataSource should refactored?